### PR TITLE
Data Refactor

### PR DESCRIPTION
**Pull request by alexflorisca**
This PR will do several things:

**Changes**
- All targeting will be set through the `targeting` property of the options object passed to `oAds.init()`
- removed the ability to validate ads traffic with the moat script. This is now being done in n-ads (https://github.com/Financial-Times/n-ads/pull/77/files#diff-240938b03dfb54ca7c6bfeaa9c3f1a7fR248), or the consuming app by simply inserting the moat script on the page.
- o-ads will not call the ads-api anymore
- `npm run test-unit` will run jest and qunit tests
- removed `ftdomdelegate` package as not being used

**TODO:**
* [x] pass `options.targeting` to `oAds.targeting.add()`
* [x] remove the `targetingApi` option
* [x] remove `dfp_targeting` option
* [x] remove `updateContext()` and add `updateTargeting()` function in main.js
* [x] require that gpt option is set with network, site and zone
* [x] remove the api module from o-ads
* [x] Update tests
* [x] Update demos
* [x] Update docs
* [x] Migration docs

### CODE REVIEW

**Comment by JakeChampion** [comment]
Hi @alexflorisca, Origami components can not have any npm dependencies, they need to be using bower dependencies only, this is because Origami supports bower users and [Build Service](https://www.ft.com/__origami/service/build/v2/) users (which uses bower).

The npm versions of Origami components is a step taking during publishing only, the reason the source code can't be written using npm components is because bower uses the git repository (the source code) as it's package registry.

**Comment by JakeChampion** [comment]
By "source code" I mean the code that exists in the Git repository. The code within the Git repository has to be written in a bower-compatible way, not using npm dependencies.

File: README.md [hunk 1]
```diff
@@ -332,13 +329,10 @@ o-ads provides some classes to add some basic branded styling to the ad slot.
 # Events
 
 ## `oAds.initialising`
-Triggered when the library starts the initialisation process. At this point in time, if targetingApi has been defined in the configuration, two separate calls are made to the targeting api in order to get ‘user’ and ‘page’ targeting parameters.
+Triggered when the library starts the initialisation process.
 
 Also at this point, if validateAdsTraffic is set to true, o-ads will check if the traffic validation script (currently moat.js) is available and use it to check if the traffic source is a valid one.
 
-## `oAds.adsAPIComplete`
```
Comment (carlesandres) [inline README.md:339]: Are we going to lose this information Alex? We are actually using it at the moment

Comment (alexflorisca) [reply README.md:339]: This has been moved to n-ads here: https://github.com/Financial-Times/n-ads/pull/77/files#diff-274b563ae51d1442f35e57d507c48e43R114

**Comment by alexflorisca** [comment]
I've reverted the changes to remove bower as per Jakes comments.

**Pull request merged**